client/src/app/components/discover-import-wizard/discover-import-wizard.tsx

```diff
@@ -129,41 +110,64 @@ const DiscoverImportWizardInner: React.FC<IDiscoverImportWizard> = ({
         }
         isVisitRequired
       >
+        {!initialPlatform ? (
+          <WizardStep
+            id="platform"
+            name={t("platformDiscoverWizard.platformSelect.stepTitle")}
+            footer={{
+              nextButtonText: t("actions.next"),
+              backButtonText: t("actions.back"),
+              isNextDisabled: platform === null,
+            }}
+          >
+            <SelectPlatform
+              platform={platform}
+              onPlatformSelected={setPlatform}
+            />
+          </WizardStep>
+        ) : null}
+
         <WizardStep
           id="filter-input"
           name={t("platformDiscoverWizard.filterInput.stepTitle")}
+          isDisabled={!platform}
           footer={{
             nextButtonText: t("actions.next"),
             backButtonText: t("actions.back"),
             isNextDisabled: !filters.isValid,
           }}
         >
-          <FilterInput
-            platform={platform}
-            onFiltersChanged={setFilters}
-            initialFilters={filters}
-          />
+          {platform ? (
+            <FilterInput
+              platform={platform}
+              onFiltersChanged={setFilters}
+              initialFilters={filters}
+            />
+          ) : null}
         </WizardStep>
 
         <WizardStep
           id="review"
           name={t("platformDiscoverWizard.review.stepTitle")}
+          isDisabled={!platform || !filters.isValid}
           footer={{
             nextButtonText: results
               ? t("actions.close")
               : t("actions.discoverApplications"),
             backButtonText: t("actions.back"),
-            isNextDisabled: !state.isReady && !results,
+            isNextDisabled: !isReady && !results,
             isBackDisabled: !!results,
             isCancelHidden: !!results,
             onNext: results ? handleCancel : onSubmitTask,
           }}
         >
-          {!results ? (
-            <Review platform={platform} filters={filters} />
-          ) : (
-            <Results results={results} />
-          )}
+          {platform ? (
+            !results ? (
+              <Review platform={platform} filters={filters} />
```

Comment:
I think it would be more readable to handle missing platform/results inside `Review` component (return null ?). This way we could just:
```
 <Review platform={platform} filters={filters}  results={results}/>
```

Reply:
I want to keep the Review and Results components separate.  I can push the null handling of platform inside the Review component.  That does make sense.